test/integration/I18n_test.rb

```diff
@@ -3,6 +3,17 @@
 
 class I18nTest < ActionDispatch::IntegrationTest
 
+  test 'should translate search input bar' do
+    available_testing_locales.each do |lang|
+      get '/change_locale/es'
+      follow_redirect!
+      get '/dashboard'
+      follow_redirect!
+      #assert_select '#searchform_input:match("placeholder",?)',I18n.t('layout._header.search')
+      assert_select '#searchform_input:match("placeholder",?)',/span/
```

Comment:
now that https://github.com/publiclab/plots2/pull/9878 is merged did you want to rebase this? Just linking relevant lines here:

https://github.com/publiclab/plots2/blob/b73594fb7d2b1a78f9f0afb074c55ac92bfd8e43/config/locales/en.yml#L631

https://github.com/publiclab/plots2/blob/b73594fb7d2b1a78f9f0afb074c55ac92bfd8e43/app/views/layouts/_header.html.erb#L20

So since we don't have a spanish equivalent, you're looking for the helper now:

https://github.com/publiclab/plots2/blob/main/config/locales/es.yml

Is it inserting this via JavaScript after the page is rendered, or in Ruby before?

Reply:
Since we're passing html = false, it should show line 167, which is just the translated string if it exists, or the English if it doesn't, right? This is all in Ruby so the only JS needed is when the placeholder is built in JavaScript. That said, are we sure the Bootstrap UI scripts (which build the placeholder) run in the limited "simulation" of a browser in integration tests? If not, best move this over to a system test as you've discussed.